### PR TITLE
[flang][driver] Fix the runtime library order

### PR DESCRIPTION
**Pull request by banach-space**
This patch makes sure that in the generated linker invocation, the input
object file, temp.o, is listed before the runtime libs:
```
ld temp.o -lFortran_main -lFortranRuntime -lFortranDecimal -lm -o a.out
```

Currently the following incorrect invocation is generated:
```
ld -lFortran_main -lFortranRuntime -lFortranDecimal -lm temp.o -o a.out
```

### CODE REVIEW

**Review by Leporacanthicus — APPROVED** [review]
LGTM

File: clang/lib/Driver/ToolChains/Gnu.cpp
Comment (jeanPerier) [inline clang/lib/Driver/ToolChains/Gnu.cpp:572]: Do you need to duplicate the `-lFortranRuntime` here ?

Comment (banach-space) [reply clang/lib/Driver/ToolChains/Gnu.cpp:572]: No, that's a leftover from debugging. Good catch, thanks! I'll update this in a few mins.

**Review by jeanPerier — APPROVED** [review]
LGTM, please confirm if I can squash and merge (or squash on your side before I merge if you'd rather).

**Comment by banach-space** [comment]
> LGTM, please confirm if I can squash and merge (or squash on your side before I merge if you'd rather).

Thanks for reviewing. Please go ahead and "squash and merge"!

**Pull request merged**
